src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -0,0 +1,115 @@
+        raise NotImplementedError()
+
+    @abstractmethod
+    def get_update_payment_status_data(
+        self, submission: Submission, options: RegistrationOptions
+    ) -> dict[str, Any]:
+        raise NotImplementedError()
+
+
+class ObjectsAPIV1Registrator(BaseObjectsAPIRegistrator):
+    """Provide the registration data for legacy (v1) registration options, using JSON templates."""
+
+    @staticmethod
+    def get_payment_context_data(submission: Submission) -> dict[str, Any]:
+        return {
+            "completed": submission.payment_user_has_paid,
+            "amount": str(submission.payments.sum_amount()),
+            "public_order_ids": submission.payments.get_completed_public_order_ids(),
+        }
+
+    @override
+    def get_record_data(
+        self,
+        submission: Submission,
+        pdf_url: str,
+        csv_url: str,
+        attachements_urls: list[str],
+        options: RegistrationOptionsV1,
+    ) -> dict[str, Any]:
+
+        context = {
+            "_submission": submission,
+            "productaanvraag_type": options["productaanvraag_type"],
+            "payment": self.get_payment_context_data(submission),
+            "variables": get_variables_for_context(submission),
+            # Github issue #661, nested for namespacing note: other templates and context expose all submission
+            # variables in the top level namespace, but that is due for refactor
+            "submission": {
+                "public_reference": submission.public_registration_reference,
+                "kenmerk": str(submission.uuid),
+                "language_code": submission.language_code,
+                "uploaded_attachment_urls": attachements_urls,
+                "pdf_url": pdf_url,
+                "csv_url": csv_url,
+            },
+        }
+
+        return render_to_json(options["content_json"], context)
+
+    @override
+    def get_update_payment_status_data(
+        self, submission: Submission, options: RegistrationOptionsV1
+    ) -> dict[str, Any]:
+        context = {
+            "variables": get_variables_for_context(submission),
+            "payment": self.get_payment_context_data(submission),
+        }
+
+        return render_to_json(options["payment_status_update_json"], context)
+
+
+REGISTRATOR_MAPPING: dict[int, BaseObjectsAPIRegistrator] = {
+    1: ObjectsAPIV1Registrator(),
+}
+
+
+def prepare_data_for_registration(
+    submission: Submission,
+    record_data: dict[str, Any],
+    objecttype: str,
+    objecttype_version: int,
+    object_mapping: MappingConfig,
+) -> dict[str, Any]:
+    """Prepare the submission data for sending it to the Objects API."""
+
+    object_data = {
+        "type": objecttype,
+        "record": {
+            "typeVersion": objecttype_version,
+            "data": record_data,
+            "startAt": get_today(),
+        },
+    }
+    apply_data_mapping(submission, object_mapping, REGISTRATION_ATTRIBUTE, object_data)
```

Comment:
this will only remain relevant for V1, in V2 we will have an explicit mapping.

We will have to figure out a migration path from the component-level "Registration" tab, but perhaps pointing to the "Variables" tab and configuration will be sufficient already.